src/stream/server/control.rs

```diff
@@ -0,0 +1,12 @@
+use crate::transfer::messages::StreamStats;
+use crate::JobId;
+use std::path::PathBuf;
+use tako::server::rpc::ConnectionDescriptor;
+use tokio::sync::oneshot;
+
+pub enum StreamServerControlMessage {
+    RegisterStream(JobId, PathBuf, oneshot::Sender<()>),
```

Comment:
Could this be a named struct? It's unclear what do these fields mean (especially the last one).

Reply:
:+1: